doc/devel/contributing.rst

```diff
@@ -0,0 +1,422 @@
+   
+      python setup.py build
+
+    every time the source code of a compiled extension is changed (for
+    instance when switching branches or pulling changes from upstream).
+
+.. _nose: http://nose.readthedocs.org/en/latest/
+.. _pep8: http://pep8.readthedocs.org/en/latest/
+
+.. note::
+    
+  **Additional dependencies for testing**: nose_ (version 1.0 or later), `mock
+  <http://www.voidspace.org.uk/python/mock/>`_ (if python < 3.3), `Ghostscript
+  <http://pages.cs.wisc.edu/~ghost/>`_, `Inkscape <http://inkscape.org>`_
+
+.. note:: To make sure the tests run locally:
+
+      * Copy setup.cfg.template to setup.cfg
+      * Edit setup.cfg to set `test` to True, and `local_freetype` to True
+      * If you have build matplotlib previously, remove the build folder.
+      * Execute the build command.
+
+When working on bleeding edge packages, setting up a
+`virtual environment
+<http://docs.python-guide.org/en/latest/dev/virtualenvs/>`_ or a `conda
+environment <http://conda.pydata.org/docs/using/envs.html>`_ is recommended.
+
+.. seealso::
+  
+  * :ref:`testing`
+
+
+Contributing code
+=================
+
+How to contribute
+-----------------
+
+The preferred way to contribute to matplotlib is to fork the `main
+repository <https://github.com/matplotlib/matplotlib/>`__ on GitHub,
+then submit a "pull request" (PR):
+
+ 1. `Create an account <https://github.com/join>`_ on
+    GitHub if you do not already have one.
+
+ 2. Fork the `project repository
+    <https://github.com/matplotlib/matplotlib>`__: click on the 'Fork' button
+    near the top of the page. This creates a copy of the code under your
+    account on the GitHub server.
+
+ 3. Clone this copy to your local disk::
+
+        $ git clone git@github.com:YourLogin/matplotlib.git
+
+ 4. Create a branch to hold your changes::
+
+        $ git checkout -b my-feature
+
+    and start making changes. Never work in the ``master`` branch!
+
+ 5. Work on this copy, on your computer, using Git to do the version
+    control. When you're done editing, do::
+
+        $ git add modified_files
+        $ git commit
+
+    to record your changes in Git, then push them to GitHub with::
+
+        $ git push -u origin my-feature
+
+Finally, go to the web page of the your fork of the matplotlib repo,
+and click 'Pull request' to send your changes to the maintainers for review.
+You may want to consider sending an email to the mailing list for more
+visibility.
+
+.. seealso::
+   
+  * `Git documentation <https://git-scm.com/documentation>`_
+  * :ref:`development-workflow`.
+  * :ref:`using-git`
+
+Contributing pull requests
+--------------------------
+
+It is recommended to check that your contribution complies with the following
+rules before submitting a pull request:
+
+  * If your pull request addresses an issue, please use the title todescribe
+    the issue and mention the issue number in the pull request description
+    to ensure a link is created to the original issue.
```

Comment:
It needs to be in the commit message or the description if you want to trigger GitHub's automatic issue closing (which only works when the PR goes to `master` _or_ when the commit (with the reference) is merged into `master`). But cross-references still work everywhere.